### PR TITLE
Add new cmdlet Reset-AzHubRouter

### PR DESCRIPTION
**Pull request by nagula-ritvika**
## Description

Expose Reset-HubRouter cmdlet to allow users to reset the routing state of the virtual hub.

Design Review Link : https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/588

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Network/Network/help/Reset-HubRouter.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
+---
```
Comment (VeryEarly) [inline src/Network/Network/help/Reset-HubRouter.md:1]: can you check why this generated help doesn't have Az prefix in cmdlet name:
Reset-HubRouter => ResetAzHubRouter

**Comment by nagula-ritvika** [comment]
> * please add and record test for new added cmdlet
> * please export new cmdlet in Network.psd1

1. This cmdlet can be used only when the RoutingState of the hub is not provisioned, which i dont think i can simulate, but i can still add it the tests. Need to wait for SDK to get merged in before i record it.
2. Done

File: src/Network/Network/help/Reset-HubRouter.md (hunk 1 above)
Comment (VeryEarly) [reply src/Network/Network/help/Reset-HubRouter.md]: Hi @nagula-ritvika ,

the name is still incorrect in this file, please regenerate a proper help markdown

**Comment by VeryEarly** [comment]
you can try pull latest changes from network-may branch, this should fix most build issue

File: src/Network/Network/help/Reset-AzHubRouter.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
+---
```
Comment (VeryEarly) [inline src/Network/Network/help/Reset-AzHubRouter.md:1]: please fix cmdlet names in this file

**Comment by VeryEarly** [comment]
- please re-record TestCortexCRUD 

- please regenerate help-markdown: Reset-AzHubRouter.md
cmdlet names are incorrect in this file

**Comment by VeryEarly** [comment]
Hi @number213 @nagula-ritvika ,

Just want to confirm, it looks like this new cmdlet, "Reset-AzHubRouter" is actually doing "Set-AzVirtualHub",

It basically get the existing VirtualHub and perform PUT operation to Overwrite the existing one.

Sorry it might be confusing since you have approved design, but this cmdlet name doesn't accurate to me, can you help me understand why using this cmdlet name?

**Comment by nagula-ritvika** [comment]
> Hi @number213 @nagula-ritvika ,
> 
> Just want to confirm, it looks like this new cmdlet, "Reset-AzHubRouter" is actually doing "Set-AzVirtualHub",
> 
> It basically get the existing VirtualHub and perform PUT operation to Overwrite the existing one.
> 
> Sorry it might be confusing since you have approved design, but this cmdlet name doesn't accurate to me, can you help me understand why using this cmdlet name?
> 
> It looks to me Reset-AzVirtualHub could be a more proper name

Hi, we have an internal Routing entity in the hub which we do not expose to the customer but we have started exposed a property called routingState which indicates the state of the routing entity. The name Reset-AzHubRouter was decided by our PMs and managers so that we have the consistent name across PS, CLI and Portal. The use of this cmdlet is to retrigger a put on the hub only when the routing state is not Provisioned. We also do not want to accidentally take in any change in the hub if the cx uses the input object and hence explicitly do a Get and then the Put.

**Comment by nagula-ritvika** [comment]
Regarding the test recording: I am hitting errors due to local build failure : can you please take a look : 
image

**Comment by VeryEarly** [comment]
> > Hi @number213 @nagula-ritvika ,
> > Just want to confirm, it looks like this new cmdlet, "Reset-AzHubRouter" is actually doing "Set-AzVirtualHub",
> > It basically get the existing VirtualHub and perform PUT operation to Overwrite the existing one.
> > Sorry it might be confusing since you have approved design, but this cmdlet name doesn't accurate to me, can you help me understand why using this cmdlet name?
> > It looks to me Reset-AzVirtualHub could be a more proper name
> 
> Hi, we have an internal Routing entity in the hub which we do not expose to the customer but we have started exposed a property called routingState which indicates the state of the routing entity. The name Reset-AzHubRouter was decided by our PMs and managers so that we have the consistent name across PS, CLI and Portal. The use of this cmdlet is to retrigger a put on the hub only when the routing state is not Provisioned. We also do not want to accidentally take in any change in the hub if the cx uses the input object and hence explicitly do a Get and then the Put.

Got it

**Comment by VeryEarly** [comment]
> Regarding the test recording: I am hitting errors due to local build failure : can you please take a look :
> image

can you try pull latest changes from network-may branch? this should fix most build issue

**Comment by nagula-ritvika** [comment]
> > Regarding the test recording: I am hitting errors due to local build failure : can you please take a look :
> > image
> 
> can you try pull latest changes from network-may branch? this should fix most build issue

Doesnt seem to be helping, tried git clean -xfd and then msbuild and still see the same errors.

**Comment by nagula-ritvika** [comment]
@VeryEarly  : updated the test recording. please take a look and merge it in :)

File: src/Network/Network/help/Reset-AzHubRouter.md (hunk 2 above)
Comment (nagula-ritvika) [reply src/Network/Network/help/Reset-AzHubRouter.md:1]: done

**Pull request merged**